### PR TITLE
[Websearch] Action rendering (v0)

### LINKED ISSUE #5190: [Multi-Actions] [Websearch] Assistant message action rendering

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Fixes #5190

Displays the action content (search query, then results) when used

This is a first version so we can start using the action & iterate. cc @Duncid at your disposal to improve it

image

image

Risk
---
N/A (gated)

Deploy Plan
---
Front

### CODE REVIEW

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 1]
```diff
@@ -1,16 +1,139 @@
...
 
-// TODO(pr,websearch) Implement this function
 export default function WebsearchAction({
   websearchAction,
 }: {
   websearchAction: WebsearchActionType;
 }) {
+  const { query } = websearchAction;
+
+  const [resultsListVisible, setResultsListVisible] = useState(false);
```
Comment (flvndvd) [inline front/components/assistant/conversation/WebsearchAction.tsx:24]: `nit`
```suggestion
  const [isResultsListVisible, setIsResultsListVisible] = useState(false);
```

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 2]
```diff
@@ -1,16 +1,139 @@
...
+
+  const { results } = websearchAction.output || { results: [] };
```
Comment (flvndvd) [inline front/components/assistant/conversation/WebsearchAction.tsx:26]:
```suggestion
  const { results } = websearchAction.output ?? { results: [] };
```

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 3]
```diff
@@ -1,16 +1,139 @@
...
+  console;
```
Comment (flvndvd) [inline front/components/assistant/conversation/WebsearchAction.tsx:27]: ✂️

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 4]
```diff
@@ -1,16 +1,139 @@
...
+                  }
+                  size="xs"
+                />
+              </Chip>
+            </div>
+          )}
+        </div>
+        <div className="col-start-2 row-span-1">
+          {!!results.length && (
+            <Transition
```
Comment (flvndvd) [inline front/components/assistant/conversation/WebsearchAction.tsx:78]: `nit`

Pretty sure you can achieve the same with className="animate-in fade-in animate-out fade-out duration-200"

File: front/components/assistant/conversation/WebsearchAction.tsx [hunk 5]
```diff
@@ -1,16 +1,139 @@
...
+                          />
+                          <div className="text-action-800">{result.title}</div>
+                        </a>
+                        <div className="flex text-xs text-element-700">
+                          <Icon
+                            visual={ExternalLinkIcon}
+                            size="xs"
+                            className="mr-1 inline-block opacity-0"
+                          />
+                          <div>{trimText(result.snippet, 80)}</div>
```
Comment (flvndvd) [inline front/components/assistant/conversation/WebsearchAction.tsx:111]: Let's use css truncate?

File: front/components/assistant/conversation/WebsearchAction.tsx (hunk 1 above)
Comment (philipperolet) [reply front/components/assistant/conversation/WebsearchAction.tsx:24]: Cargo Culting m'a tuer

File: front/components/assistant/conversation/WebsearchAction.tsx (hunk 2 above)
Comment (philipperolet) [reply front/components/assistant/conversation/WebsearchAction.tsx:26]: Cargo Culting m'a tuer

File: front/components/assistant/conversation/WebsearchAction.tsx (hunk 4 above)
Comment (philipperolet) [reply front/components/assistant/conversation/WebsearchAction.tsx:78]: Cargo Culting m'a tuer

**Review by Duncid — APPROVED** [review]

**Comment by philipperolet** [comment]
@flvndvd Adressed your coms :+1:

**Pull request merged**